src/main/java/com/minecolonies/coremod/client/gui/WindowHutBuilder.java

```diff
@@ -64,9 +124,8 @@ public String getBuildingName()
     public void onOpened()
     {
         super.onOpened();
-        pullResourcesFromHut();
 
-        final Object[] entries = resources.entrySet().toArray();
+        pullResourcesFromHut();
 
         final ScrollingList resourceList = findPaneOfTypeByID(LIST_RESOURCES, ScrollingList.class);
         resourceList.setDataProvider(new ScrollingList.DataProvider()
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR') Reduce this anonymous class number of lines from 48 to at most 30, or make it a named class. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://home.kk-sc.de:9000/coding_rules#rule_key=squid%3AS1188)